pairing-tool/index.js

```diff
@@ -1,108 +1,101 @@
+        resolve()
+      } else {
+        const unitQuestion = {
+          type: 'input',
+          name: 'unit',
+          message: 'What is the unit for this new button?',
+        }
+
+        const phoneNumberQuestion = {
+          type: 'input',
+          name: 'phoneNumber',
+          message: 'What is the phone number for this new button?',
+        }
+
+        const confirmQuestion = {
+          type: 'confirm',
+          name: 'confirmation',
+          message: 'Create a button with the info above?',
+        }
+
+        const response = await prompt([unitQuestion, phoneNumberQuestion, confirmQuestion])
 
-    // eslint-disable-next-line no-unused-vars -- the reject var is included for clarity
-    let promise = new Promise((resolve, reject) => {
- 
-        let wizard = new FlicScanWizard()
-        // wizard.on('foundPublicButton', (bdAddr, name) => console.log(`Found a public button with UUID ${bdAddr}, keep holding it down...`))
-        // wizard.on('buttonConnected', (bdAddr, name) => console.log(`Connected to a button with UUID ${bdAddr}, keep holding it down...`)) 
-
-        // eslint-disable-next-line no-unused-vars -- these vars may be useful soon
-        wizard.on('completed', async (result, bdAddr, name) => {
-            console.log(`Completed pairing for button ${bdAddr} with result: ${result}`)
-            if(result !== 'WizardSuccess') {
-                resolve()
-            }
-            else {
-
-                let unitQuestion = {
-                    type: 'input',
-                    name: 'unit',
-                    message: 'What is the unit for this new button?'
-                }
-
-                let phoneNumberQuestion = {
-                    type: 'input',
-                    name: 'phoneNumber',
-                    message: 'What is the phone number for this new button?'
-                }
-
-                let confirmQuestion = {
-                    type: 'confirm',
-                    name: 'confirmation',
-                    message: 'Create a button with the info above?'
-                }
-
-                const response = await prompt([unitQuestion, phoneNumberQuestion, confirmQuestion])
-                
-                if(response.confirmation) {
-                    // eslint-disable-next-line no-unused-vars -- these vars may be useful soon
-                    client.getButtonInfo(bdAddr, (bdAddr, uuid, color, serialNumber) => {
-                        csvWriter.write([uuid, response.unit, response.phoneNumber, serialNumber])
-                        resolve()
-                    })
-                }
-                else {
-                    resolve()
-                }
-            }
-        })
-
-        client.addScanWizard(wizard)
+        if (response.confirmation) {
+          // eslint-disable-next-line no-unused-vars -- these vars may be useful soon
+          client.getButtonInfo(bdAddr, (uuid, color, serialNumber) => {
```

Comment:
> Thanks for doing this.
> 
> Would you mind waiting until after the PR for promise rejection warnings is merged? There will be merge conflicts, but only 4 files.

No problem :)

Reply:
Funny, it seems you replied to my upper-comment on this one, instead of the comment itself. It seems to have removed a parameter to the inner arrow-function.

Reply:
Oops, I think it was because I was looking at it in the overall PR view as opposed to per file (which is what I'm doing now haha)

Reply:
If I put that back it says "`bdAddr` is declared but its value is never read" and also "`bdAddr` is already declared in the upper scope"

There are a bunch of places in these files where extra things are passed in with comments like "this might be useful one day" and are not used, the possible fixes it suggests for this one when I put that back are to prefix it with an underscore or to disable `no-shadow`, for now I'm going to do the underscore thing because otherwise Travis will fail when I push my updates

Reply:
I think something is still needed there. You could call it something else if you wanted, like `bdAddress` to get rid of the first error message. The reason I think that it's a problem to remove it is that it looks like it's being used as a callback by `client.getButtonInfo`. So somewhere in `client.getButtonInfo` (which third-party code written by Flic), it is going to call the callback with four parameters in the order  `(bdAddr, uuid, color, serialNumber)`. If we define our callback with only three parameters `(uuid, color, serialNumber)`, then I think it will just silently assign the value that semantically is `bdAddr` to our variable `uuid`, and the value that is semantically `uuid` to our variable `color`, and so on. Does that make sense?

Reply:
oooh ok, that does make sense yes, it wouldn't know what external things need/are using. It should be fine in that case with the _ prepended to the name I think!